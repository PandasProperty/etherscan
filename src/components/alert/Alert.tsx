import AlerProps from "./types";

const ICONS = {
  ERROR: (
    <div className="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-red-500 bg-red-100 rounded-lg dark:bg-red-800 dark:text-red-200">
      <svg className="w-5 h-5" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 20 20">
        <path d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 11.793a1 1 0 1 1-1.414 1.414L10 11.414l-2.293 2.293a1 1 0 0 1-1.414-1.414L8.586 10 6.293 7.707a1 1 0 0 1 1.414-1.414L10 8.586l2.293-2.293a1 1 0 0 1 1.414 1.414L11.414 10l2.293 2.293Z"/>
      </svg>
    </div>
  ),
  SUCCESS: (
    <div className="inline-flex items-center justify-center flex-shrink-0 w-8 h-8  text-sm text-green-800 border border-green-300 rounded-lg bg-green-50 dark:bg-gray-800 dark:text-green-400 dark:border-green-800" role="alert">
      <svg className="flex-shrink-0 inline w-4 h-4" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 20 20">
        <path d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5ZM9.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3ZM12 15H8a1 1 0 0 1 0-2h1v-3H8a1 1 0 0 1 0-2h2a1 1 0 0 1 1 1v4h1a1 1 0 0 1 0 2Z"/>
      </svg>
    </div>
  )
}
export default function ErrorToast({
  message,
  type,
  dataTestId,
  onClose
}: AlerProps) {
  return (
    <div data-testid={dataTestId} className="flex items-center w-full max-w-lg p-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800">
      {ICONS[type]}
      <div className="ms-3 text-sm font-normal" data-testid={`${dataTestId}-text`}>{message}</div>
      {
        onClose && (
          <button data-testid={`${dataTestId}-button`} type="button" onClick={onClose} className="ml-3.5 rounded-lg focus:ring-2 focus:ring-gray-300hover:b g-gray-100 inline-flex items-center justify-center h-8 w-8 text-gray-500 hover:text-white bg-gray-800 hover:bg-gray-700" data-dismiss-target="#toast-danger" aria-label="Close">
            <span className="sr-only">Close</span>
            <svg className="w-3 h-3" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 14 14">
              <path stroke="currentColor" strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"/>
            </svg>
          </button>
        )
      }
    </div>
  );
}
