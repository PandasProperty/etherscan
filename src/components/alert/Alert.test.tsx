import "@testing-library/jest-dom"

import { act, fireEvent, render, waitFor } from "@testing-library/react";

import Alert from ".";

describe("Alert", () => {
  it("matched snapshot", () => {
    const onClose = jest.fn();
    const { asFragment, getByTestId, queryByTestId } = render(<Alert dataTestId="TestId" message="Dummy alert" type="ERROR" onClose={onClose} />);
    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId("TestId")).toBeInTheDocument();

    act(() => {
      fireEvent.click(getByTestId("TestId-button"));
    });

    waitFor(() => {
      expect(queryByTestId("TestId")).not.toBeInTheDocument();
    });
  });
});
