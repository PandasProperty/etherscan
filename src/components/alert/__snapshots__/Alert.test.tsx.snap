// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matched snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full max-w-lg p-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
    data-testid="TestId"
  >
    <div
      class="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-red-500 bg-red-100 rounded-lg dark:bg-red-800 dark:text-red-200"
    >
      <svg
        aria-hidden="true"
        class="w-5 h-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 11.793a1 1 0 1 1-1.414 1.414L10 11.414l-2.293 2.293a1 1 0 0 1-1.414-1.414L8.586 10 6.293 7.707a1 1 0 0 1 1.414-1.414L10 8.586l2.293-2.293a1 1 0 0 1 1.414 1.414L11.414 10l2.293 2.293Z"
        />
      </svg>
    </div>
    <div
      class="ms-3 text-sm font-normal"
      data-testid="TestId-text"
    >
      Dummy alert
    </div>
    <button
      aria-label="Close"
      class="ml-3.5 rounded-lg focus:ring-2 focus:ring-gray-300hover:b g-gray-100 inline-flex items-center justify-center h-8 w-8 text-gray-500 hover:text-white bg-gray-800 hover:bg-gray-700"
      data-dismiss-target="#toast-danger"
      data-testid="TestId-button"
      type="button"
    >
      <span
        class="sr-only"
      >
        Close
      </span>
      <svg
        aria-hidden="true"
        class="w-3 h-3"
        fill="none"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
